### DISCUSSION TITLE
Ubuntu base / slim

### DISCUSSION THREAD

**Discussion by FredsterCode** [Joshua-Riek/ubuntu-rockchip#641] [Ideas]
First and foremost Thanks. I'm running Ubuntu server on the orange pi 5 and it's rock solid (pun is accidental). 
I'm looking for a slimmer footprint though. Not quite Alpine Linux slim but less all the same. 
Security exposure and updates are something to consider but my main interest is resources and image backup size. 
Could you consider making a closer to base image of the server? 
Obviously keep your efforts for the SBC and openssh. There are almost certainly a few other packages that make sense to keep but probably only a few. 
Samba and the cloud stuff for example can be installed later if people want it. 
Assuming a device is close or connected to the internet then we can fill it up with tools and toys after install.
I looked into ripping things out of the preinstalled server image and got scared. I might manage it once without breaking things but doing it right after each new release is daunting.
I confess I don't know the effort involved in creating the images already being published but I thought if I don't make the suggestion then it's even less likely to happen. 
Alternatively has anyone got a "slim down Ubuntu " script that's reliable? Or other ideas? 
Thanks.

**Comment by Joshua-Riek** [comment]
Sadly Ubuntu is becoming more bloated as time goes on. I may not be able to host downloads on GitHub for Ubuntu 24.04 due to how large the desktop image will be about 2GB.

Now I could slim things down, but it would require me to disable a bunch of drivers in the kernel and remove WiFi + Bluetooth firmware. I can make a Ubuntu image in under 1GB, but it will be missing a lot of drivers some users may require, the primary issue are WiFi and Bluetooth drivers and firmware, they take a lot of space.

**Reply by FredsterCode** [reply]
I don't know how Ubuntu has the drivers bundled but could they be rebundled to only have the various arm ones? The effort in maintaining that for each release is possibly not attractive either. 
As another approach. Can the currently used drivers on any given install be identified and the additional ones be deleted? 

As another approach altogether. Is there a toolkit that makes it relatively easy for us to select a target device and in some way build Ubuntu for that device? 
That sounds very Arch which was not my intention.

Perhaps Debian is the way forward and then build up from that. Currently your efforts are on Ubuntu so I don't expect the Debian options to be as solid.

**Reply by Joshua-Riek** [reply]
> As another approach altogether. Is there a toolkit that makes it relatively easy for us to select a target device and in some way build Ubuntu for that device?
That sounds very Arch which was not my intention.

Ubuntu has a program called `ubuntu-image` to do this. But the documentation does exist and you can not customize anything without hacking live-build.
